### PR TITLE
Make JustSayingBus.Start async

### PR DESCRIPTION
**Pull request by maisiesadler**

### CODE REVIEW

File: src/JustSaying/Messaging/Channels/ConsumerBus.cs [hunk 1]
```diff
@@ -57,9 +57,12 @@ public Task Run(CancellationToken stoppingToken)
 
             _logger.LogInformation("Consumer bus successfully started");
 
-            return Task.WhenAll(completionTasks);
+            Completion = Task.WhenAll(completionTasks);
+            return Completion;
         }
 
+        public Task Completion { get; private set; }
```
Comment (gkinsman) [inline src/JustSaying/Messaging/Channels/ConsumerBus.cs]: Do we need this? `ConsumerBus.Start` returns the completion event already, so we could just return that instead?

File: tests/JustSaying.UnitTests/JustSayingBus/WhenRegisteringMessageHandlers.cs [hunk 2]
```diff
@@ -29,14 +30,16 @@ protected override void Given()
...
+        protected override async Task WhenAsync()
         {
             SystemUnderTest.AddQueue(_region, _queue);
             SystemUnderTest.AddMessageHandler(_futureHandler1);
             SystemUnderTest.AddMessageHandler(_futureHandler2);
-            SystemUnderTest.Start();
 
-            return Task.CompletedTask;
+            var cts = new CancellationTokenSource();
+            cts.CancelAfter(TimeSpan.FromMilliseconds(100));
```
Comment (gkinsman) [inline tests/JustSaying.UnitTests/JustSayingBus/WhenRegisteringMessageHandlers.cs]: is there a base timeout prop we can use?

File: src/JustSaying/Messaging/Channels/ConsumerBus.cs (hunk 1 above)
Comment (maisiesadler) [reply src/JustSaying/Messaging/Channels/ConsumerBus.cs]: It's for when the ConsumerBus has already been created/started in JustSayingBus. We wouldn't want to call Run again.
The other option is to use the same pattern in ConsumerBus as in the JustSayingBus where if it's already started we can return the completion task? So just do this internally?

Comment (gkinsman) [reply src/JustSaying/Messaging/Channels/ConsumerBus.cs]: Ah right, yeah I think we can just maintain a private `Task _completion` variable that is returned to callers. Not sure we need to put it in the public interface if its just for tracking starts?

**Pull request merged**